src/Context/FormContext.tsx

```diff
@@ -0,0 +1,49 @@
+import { createContext, useContext, useReducer, ReactNode, Dispatch } from 'react'; 
+
+interface FormState {
+    [key: string]: any; // example of the interface for now // 
+}
+
+interface FormAction {   
+    type: string; 
+    payload?: any;      
+}
+
+interface FormProviderProps {
+    children: ReactNode; 
+}
+
+interface FormContextProps { 
+  formState: FormState; // wartości które przechowuje context, 
```

Comment:
Please do not leave comments in Polish.

Also - it's gonna be the React Hook Form that is gonna be holding the state of the form, so you don't need it here